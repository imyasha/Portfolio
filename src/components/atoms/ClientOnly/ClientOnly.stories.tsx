import { ClientOnly } from './ClientOnly';

export default {
	title: 'atoms/ClientOnly',
	component: ClientOnly,
};

interface Props {
	children: React.ReactNode;
}

export const ClientOnlyStory = ({ children }: Props) => (
	<ClientOnly>{children}</ClientOnly>
);

ClientOnlyStory.storyName = 'ClientOnly';
ClientOnlyStory.args = {
	children: 'Content only gets rendered on the client',
};
