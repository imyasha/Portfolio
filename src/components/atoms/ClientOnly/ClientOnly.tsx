import { FC, ReactNode, useEffect, useState } from 'react';

interface ClientOnlyProps {
	children: ReactNode;
}

export const ClientOnly: FC<ClientOnlyProps> = ({ children, ...delegated }) => {
	const [hasMounted, setHasMounted] = useState(false);

	useEffect(() => {
		setHasMounted(true);
	}, []);

	if (!hasMounted) {
		return null;
	}

	return <div {...delegated}>{children}</div>;
};
