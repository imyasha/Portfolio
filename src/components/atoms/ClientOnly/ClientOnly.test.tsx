import { ClientOnly } from '../ClientOnly';
import { render } from '@testing-library/react';

describe('ClientOnly', () => {
	it('renders correctly', () => {
		const { container } = render(
			<ClientOnly>Client-side only content</ClientOnly>
		);
		expect(container).toMatchSnapshot();
	});
});
