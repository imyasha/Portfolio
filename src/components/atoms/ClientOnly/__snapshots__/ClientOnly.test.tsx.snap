// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientOnly renders correctly 1`] = `
<div>
  <div>
    Client-side only content
  </div>
</div>
`;
